staging/src/k8s.io/kubectl/pkg/describe/describe.go

```diff
@@ -3861,13 +3873,77 @@ func describeNode(node *corev1.Node, nodeNonTerminatedPodsList *corev1.PodList,
 		} else {
 			w.Write(LEVEL_0, "Pods:\tnot authorized\n")
 		}
+
 		if events != nil {
 			DescribeEvents(events, w)
 		}
 		return nil
 	})
 }
 
+// Add helper function to describe ResourceSlices
+func describeNodeResourceSlices(w PrefixWriter, node *corev1.Node, resourceSlices []resourcev1beta1.ResourceSlice) {
+	w.Write(LEVEL_0, "ResourceSlices:\n")
+	w.Write(LEVEL_0, "  Name\t\tDriver\t\tPool\t\tAssociation\n")
+	w.Write(LEVEL_0, "  ----\t\t------\t\t----\t\t-----------\n")
+
+	// Sort slices by name for consistent output
+	sort.Slice(resourceSlices, func(i, j int) bool {
+		return resourceSlices[i].Name < resourceSlices[j].Name
+	})
+
+	for _, slice := range resourceSlices {
+		association := ""
+		// Determine association type
+		switch {
+		case slice.Spec.NodeName != "" && slice.Spec.NodeName == node.Name:
+			association = "Direct node match"
+		case slice.Spec.AllNodes:
+			association = "Available to all nodes"
+		case slice.Spec.NodeSelector != nil:
+			// Check if node selector matches
+			nodeSelector, err := nodeaffinity.NewNodeSelector(slice.Spec.NodeSelector)
+			if err != nil {
+				continue
+			}
+			if nodeSelector.Match(node) {
+				association = "Via node selector"
+			}
+		case slice.Spec.PerDeviceNodeSelection == ptr.To(true):
+			// Check if any devices are assigned to this node
+			hasDevices := false
+			for _, device := range slice.Spec.Devices {
+				if device.Basic.NodeName != nil && *device.Basic.NodeName == node.Name {
+					hasDevices = true
+					break
+				}
+			}
+			if hasDevices {
+				association = "Per-device selection"
+			}
+		}
+
+		// Only show slices associated with this node
+		if association != "" {
+			w.Write(LEVEL_0, "  %s\t\t%s\t\t%s\t\t%s\n",
+				slice.Name,
+				slice.Spec.Driver,
+				slice.Spec.Pool.Name,
+				association)
```

Comment:
v1beta2 will have "Devices" rather than the Basic/Composite oneOf. So, I think you can display a count of the devices here too.